wepool-app/src/screens/MatchedDrivers.tsx

```diff
@@ -1,54 +1,82 @@
+import { useQuery } from "@apollo/client";
+import { Oops } from '../components/Oops';
+
+// queries
+import GetOpenRides from "../queries/GET/RideQueries";
 
 export const MatchedDrivers = ({navigation}: RootStackScreenProps<'MatchedDrivers'>) => {
 
     /**
-     * Getting dummy users info for testing purposes. 
-     * TODO: 
-     *  - Integrate with backend
+     * TODO:
      *  - A varible to know user type will be needed
+     *  - Apply 'Star' ratings
+     *  - Variables missing: date, time, money, notes
      */
+    
+    //open and close ride detail modal
+    const [openDetails, setOpenDetails] = useState(false);
+    const [cardId, setCardId] = useState(-1);
+    
+    function handleOpenDetails(){
+        setOpenDetails(!openDetails)
+    }
+    function handleCardId(id: number){
+        setCardId(id)
+    }
+    const { loading, error, data } = useQuery(GetOpenRides);
 
-    const [allUsers, setUser] = useState<User[]>();
+    const [openRides, setOpenRides] = useState<Ride[] | null>(null);
     useEffect(() => {
-        setUser(_testUsers)
-    }, []);
-        
-    const [openDetails, setOpenDetails] = useState(false); //open and close ride detail modal
-    function handleOnPressDetails(){
-        setOpenDetails(!openDetails)
+      if (data && data.rides) setOpenRides(data.rides);
+    }, [loading]);
+  
+    if (error)
+      console.log([JSON.stringify({ data }), error, error.networkError]);
+    else if (loading || !openRides) {
+      console.log("Loading...");
+      return (
+        <View>
+          <Text>Loading...</Text>
+        </View>
+      );
     }
 
     return (
         <View style = {styles.container}>
             <View style = {styles.headerContainer}>
-                <HeaderBar user={allUsers ? allUsers[0].fname : 'Test'} userType="Rider"/>
+                <HeaderBar user={'Test'} userType="Rider"/>
             </View>
             <View style = {styles.contentContainer}>
                 <View style = {styles.backButton}>
                     <BackButton onPress={() => navigation.goBack()} />
                 </View>
                 <Header text="Matched Drivers"/>
                 <View style = {styles.cardsContainer}>
-                    <ScrollView>
-                    {_testUsers.map((user) => {
-                        return (
-                            <View key={user.id} style = {styles.card}>
-                                <DriverCard date='20 Apr' time='08:00' start_loc={user.street} final_loc={user.city} driverName = {user.fname} status={true} handleOnPressDetails = {handleOnPressDetails}/>
-                            </View>
-                        );
-                    })}
-                    </ScrollView>
-                    <RideDetailsModal openDetails = {openDetails} handleOnPressDetails = {handleOnPressDetails}/>
+                    {openRides ? (
+                        <ScrollView>
+                                {openRides.map((ride) => 
```

Comment:
@BernardoGZ Cleaned this up to only use the map method, the check you were performing on the array (if it's length was greater than 1) is unneccesary once you verify that the array exists. That is because if we verify that the array is nonnull (as we are doing on line 63 in this case) we can use `Array.map()` without issues, since accessing that method on an empty array will only produce an empty array.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
Nice, maybe the error showed before checking the not-null openRides. Thanks for the update.